### PR TITLE
add unit tests for server.py

### PR DESCRIPTION
**Pull request by destefy**

### CODE REVIEW

**Comment by anandj91** [comment]
Can we update the README with instructions on running the tests?

**Comment by anandj91** [comment]
can you also create a `requirements-dev.txt` and add `pytest` package as a dependency?

**Comment by destefy** [comment]
> can you also create a `requirements-dev.txt` and add `pytest` package as a dependency?

you read my mind.
jsut to be clear you want a seperate requirements file for development? Should I make a second setup.py type file or just include instruction to also install requirements-dev.txt

**Review by anandj91 — CHANGES_REQUESTED** [review]

File: tests/pytest.ini [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# pytest.ini
+[pytest]
+env =
+    CENTML_CACHE_DIR=/home/ubuntu/.cache/centml-test
```
[the remark this answers is not in the document]
Comment (destefy) [reply tests/pytest.ini:4]: will do

**Comment by destefy** [comment]
Note to self: check if there is a way to specify a list of test cases in a file or something for local testing

File: centml/compiler/server.py [hunk 2]
```diff
@@ -86,12 +86,12 @@ async def compile_model_handler(model_id: str, model: UploadFile, inputs: Upload
...
     compiled_forward_path = os.path.join(storage_path, model_id, "cgraph.zip")
     if not os.path.isfile(compiled_forward_path):
-        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Download: compiled file not found")
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="Download: compiled file not found")
     return FileResponse(compiled_forward_path)
 
 
 def run():
-    uvicorn.run("server:app", host=config_instance.SERVER_IP, port=int(config_instance.SERVER_PORT))
+    uvicorn.run(app, host=config_instance.SERVER_IP, port=int(config_instance.SERVER_PORT))
```
Comment (anandj91) [inline centml/compiler/server.py:94]: Can you rebase your branch to the latest main? This fix was already merged to main. No need to do it again.

File: requirements.txt [hunk 3]
```diff
@@ -5,6 +5,3 @@ python-multipart>=0.0.6
 black>=23.10.0
 pylint>=3.0.1
 Requests==2.31.0
-
```
Comment (anandj91) [inline requirements.txt:8]: I checked your git history. It looks like you merged the main branch with your branch. Try to do rebase instead.

**Review by anandj91 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 3 above)
Comment (anandj91) [reply requirements.txt:8]: you might need to do some git history rewriting. Let me know if you need help.

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+def pytest_addoption(parser):
+    parser.addoption("--sanity", action="store_true", help="Run sanity tests (exclude 'gpu' tests)")
+
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--sanity"):
+        skip_gpu = pytest.mark.skip(reason="skipping gpu tests for sanity run")
```
Comment (anandj91) [inline tests/conftest.py:8]: "Skipping GPU tests for sanity run"

File: tests/pytest.ini [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# pytest.ini
+[pytest]
+env =
+    CENTML_CACHE_DIR=/tmp/centml-test
+filterwarnings = 
+    ignore::UserWarning
+markers =
```
Comment (anandj91) [inline tests/pytest.ini:7]: what is this for?

File: tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+        for item in items:
+            if "gpu" in item.keywords:
+                item.add_marker(skip_gpu)
```
Comment (anandj91) [inline tests/conftest.py:11]: Add a new line at the end of the file.
Ideally, the linter should catch issues like this. Can you please check?

**Review by anandj91 — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 6 above)
Comment (destefy) [reply tests/conftest.py:11]: Yep you're right. Not sure why this wasn't caught.

**Pull request merged**